### ISSUE TITLE
SplitFMNIST error when enumerating the dataset

### ISSUE THREAD

**Issue by AndreaCossu** [ContinualAI/avalanche#490]
On `SplitFMnist` is not possible to loop over the experience dataset.

To reproduce:
```
from avalanche.benchmarks import SplitFMNIST

scenario = SplitFMnist(5)
dataset = scenario.train_stream[0].dataset
for el in dataset:
  pass
```

returns 

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/cossu/avalanche/avalanche/benchmarks/utils/avalanche_dataset.py", line 306, in __getitem__
    return TupleTLabel(manage_advanced_indexing(
  File "/home/cossu/avalanche/avalanche/benchmarks/utils/dataset_utils.py", line 320, in manage_advanced_indexing
    single_element = single_element_getter(int(single_idx))
  File "/home/cossu/avalanche/avalanche/benchmarks/utils/avalanche_dataset.py", line 1035, in _get_single_item
    return self._process_pattern(self._dataset[idx], idx)
  File "/home/cossu/miniconda3/envs/avalanche-env/lib/python3.9/site-packages/torch/utils/data/dataset.py", line 272, in __getitem__
    return self.dataset[self.indices[idx]]
  File "/home/cossu/avalanche/avalanche/benchmarks/utils/avalanche_dataset.py", line 306, in __getitem__
    return TupleTLabel(manage_advanced_indexing(
  File "/home/cossu/avalanche/avalanche/benchmarks/utils/dataset_utils.py", line 320, in manage_advanced_indexing
    single_element = single_element_getter(int(single_idx))
  File "/home/cossu/avalanche/avalanche/benchmarks/utils/avalanche_dataset.py", line 659, in _get_single_item
    return self._process_pattern(self._dataset[idx], idx)
  File "/home/cossu/avalanche/avalanche/benchmarks/utils/avalanche_dataset.py", line 306, in __getitem__
    return TupleTLabel(manage_advanced_indexing(
  File "/home/cossu/avalanche/avalanche/benchmarks/utils/dataset_utils.py", line 320, in manage_advanced_indexing
    single_element = single_element_getter(int(single_idx))
  File "/home/cossu/avalanche/avalanche/benchmarks/utils/avalanche_dataset.py", line 1035, in _get_single_item
    return self._process_pattern(self._dataset[idx], idx)
  File "/home/cossu/avalanche/avalanche/benchmarks/utils/dataset_utils.py", line 207, in __getitem__
    result = super().__getitem__(idx)
  File "/home/cossu/avalanche/avalanche/benchmarks/utils/dataset_utils.py", line 184, in __getitem__
    return self.dataset[self.indices[idx]]
  File "/home/cossu/avalanche/avalanche/benchmarks/utils/avalanche_dataset.py", line 306, in __getitem__
    return TupleTLabel(manage_advanced_indexing(
  File "/home/cossu/avalanche/avalanche/benchmarks/utils/dataset_utils.py", line 320, in manage_advanced_indexing
    single_element = single_element_getter(int(single_idx))
  File "/home/cossu/avalanche/avalanche/benchmarks/utils/avalanche_dataset.py", line 659, in _get_single_item
    return self._process_pattern(self._dataset[idx], idx)
  File "/home/cossu/avalanche/avalanche/benchmarks/utils/avalanche_dataset.py", line 306, in __getitem__
    return TupleTLabel(manage_advanced_indexing(
  File "/home/cossu/avalanche/avalanche/benchmarks/utils/dataset_utils.py", line 320, in manage_advanced_indexing
    single_element = single_element_getter(int(single_idx))
  File "/home/cossu/avalanche/avalanche/benchmarks/utils/avalanche_dataset.py", line 659, in _get_single_item
    return self._process_pattern(self._dataset[idx], idx)
  File "/home/cossu/avalanche/avalanche/benchmarks/utils/avalanche_dataset.py", line 669, in _process_pattern
    pattern, label = self._apply_transforms(pattern, label)
  File "/home/cossu/avalanche/avalanche/benchmarks/utils/avalanche_dataset.py", line 680, in _apply_transforms
    pattern = self.transform(pattern)
  File "/home/cossu/miniconda3/envs/avalanche-env/lib/python3.9/site-packages/torchvision/transforms/transforms.py", line 67, in __call__
    img = t(img)
  File "/home/cossu/miniconda3/envs/avalanche-env/lib/python3.9/site-packages/torch/nn/modules/module.py", line 727, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/cossu/miniconda3/envs/avalanche-env/lib/python3.9/site-packages/torchvision/transforms/transforms.py", line 226, in forward
    return F.normalize(tensor, self.mean, self.std, self.inplace)
  File "/home/cossu/miniconda3/envs/avalanche-env/lib/python3.9/site-packages/torchvision/transforms/functional.py", line 284, in normalize
    tensor.sub_(mean).div_(std)
RuntimeError: output with shape [1, 32, 32] doesn't match the broadcast shape [3, 32, 32]
```

**Comment by AndreaCossu** [comment]
The problem seems to be related to the transformation, which were copied from the ones of CIFAR. Since FMNIST is a one-channel image of 28x28, they generated an error. I'm now creating a PR to solve this issue.

**Comment by AndreaCossu** [comment]
Closed by #529
